indico/modules/events/registration/controllers/display.py

```diff
@@ -88,8 +82,8 @@ class RHRegistrationFormList(RHRegistrationFormDisplayBase):
 
     def _process(self):
         regforms = RegistrationForm.find_all(RegistrationForm.is_scheduled, event_id=int(self.event.id))
-        if _can_redirect_to_single_regform(regforms):
-            return redirect(url_for('.display_regform_summary', regforms[0]))
+        if len(regforms) == 1 and regforms[0].can_submit(session.user):
```

Comment:
Yeah, this is basically useless, as the button link already points to the list or a single form according to how many there are.